lib/third-party-resources.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+const ApiGroup = require('./api-group');
+const BaseObject = require('./base');
+
+class ThirdPartyResource extends ApiGroup {
+  constructor(options) {
```

Comment:
Maybe not for this PR, but it might be a good user experience if one could pass in an array of existing third party resources to the constructor and have the object created here rather than users having to invoke `addResource` on every third party resource themselves.